extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/sharing/user/SharedAuthenticatedUser.java

```diff
@@ -99,7 +102,57 @@ public String getIdentifier() {
 
     @Override
     public void setIdentifier(String identifier) {
-        throw new UnsupportedOperationException("Users authenticated via share keys are immutable.");
+        throw new UnsupportedOperationException(
+                "Users authenticated via share keys are immutable.");
+    }
+
+    @Override
+    public ObjectPermissionSet getActiveConnectionPermissions()
+            throws GuacamoleException {
+        throw new UnsupportedOperationException(
+                "Shared users do not have any assigned permissions.");
+    }
```

Comment:
The `ObjectPermissionSet` representation of no permissions is an implementation of `ObjectPermissionSet` which returns no permissions, such as [`ObjectPermissionSet.EMPTY_SET`](http://guacamole.apache.org/doc/guacamole-ext/org/apache/guacamole/net/auth/permission/ObjectPermissionSet.html#EMPTY_SET).

Retrieving user permissions is a common operation and needs to succeed. Throwing `UnsupportedOperationException` doesn't indicate that the user has no permissions, but rather that permission retrieval is not supported, and will cause an internal error whenever any part of the webapp attempts to retrieve the permissions of this user.

Reply:
Yeah, this was one of my many attempts to figure out some elegant or sexy way to do this.  I'll refactor it, here, soon.